metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/resolver/ClasspathFunctionResolver.java

```diff
@@ -0,0 +1,179 @@
+ * part of Apache Metron.
+ *
+ *   stellar.function.resolver.includes = org.apache.metron.*
+ *
+ * The following property definition will exclude Stellar functions that are part of
+ * Metron's management suite of function.
+ *
+ *   stellar.function.resolver.excludes = org.apache.metron.management.*
+ *
+ * The following property definition would also exclude the Stellar functions that are
+ * part of the management suite of functions.  Of course, this may also exclude other
+ * packages, but this serves as an example of the types of expression that can be used.
+ *
+ *   stellar.function.resolver.excludes = org\\.management.*
+ *
+ */
+public class ClasspathFunctionResolver extends BaseFunctionResolver {
+
+  /**
+   * The key for a global property that defines one or more regular expressions
+   * that specify what should be included when searching for Stellar functions.
+   */
+  protected static final String STELLAR_SEARCH_INCLUDES_KEY = "stellar.function.resolver.includes";
+
+  /**
+   * The key for a global property that defines one or more regular expressions
+   * that specify what should be excluded when searching for Stellar functions.
+   */
+  protected static final String STELLAR_SEARCH_EXCLUDES_KEY = "stellar.function.resolver.excludes";
+
+  /**
+   * The includes and excludes can include a list of multiple includes or excludes that
+   * are delimited by these values.
+   */
+  protected static final String STELLAR_SEARCH_DELIMS = "[,:]";
+
+  /**
+   * Regular expressions defining packages that should be included in the Stellar function resolution
+   * process.
+   */
+  private List<String> includes;
+
+  /**
+   * Regular expressions defining packages that should be excluded from the Stellar function resolution
+   * process.
+   */
+  private List<String> excludes;
+
+  public ClasspathFunctionResolver() {
+    this.includes = new ArrayList<>();
+    this.excludes = new ArrayList<>();
+  }
+
+  /**
+   * Includes one or more packages in the Stellar function resolution process.  The packages
+   * to include can be specified with a regular expression.
+   * @param toInclude The regular expressions.
+   */
+  public void include(String... toInclude) {
+    for(String incl : toInclude) {
+      includes.add(incl);
+    }
+  }
+
+  /**
+   * Excludes one or more packages from the Stellar function resolution process.  The packages
+   * to exclude can be specified with a regular expression.
+   * @param toExclude The regular expressions defining packages that should be excluded.
+   */
+  public void exclude(String... toExclude) {
+    for(String excl : toExclude) {
+      excludes.add(excl);
+    }
+  }
+
+  @Override
+  public void initialize(Context context) {
+    super.initialize(context);
+
+    // do we have access to the global configuration?
+    if(context != null) {
+
+      Optional<Object> optional = context.getCapability(GLOBAL_CONFIG, false);
```

Comment:
While I think it's a sensible thing to pull these from the global config on a running cluster, on the situation where the REPL is used, it is likely that you may want to inject or ignore the includes and excludes from the global config, rather than inherit them.  Can we perhaps not couple function resolution with the global config?  I'd recommend it be a separate capability name called `STELLAR_CONFIG`.  Where we pull that capability from should depend on the context (e.g. in the REPL or MR jobs it should be probably passed in via some external parameters, in the running cluster it'll probably be in the global config or maybe a subsection of the global config like `globalConfig.get("stellarConfig")`.